AccountingSystem/Client/DataStoreClient.py

```diff
@@ -58,7 +55,13 @@ def addRegister( self, register ):
       return retVal
     if gConfig.getValue( '/LocalSite/DisableAccounting', False ):
       return S_OK()
+    
+    # a lock below can only prevent to send the register 
+    # with an 'earlier' commit what normally is not a problem at all
```

Comment:
Please consider this part in context of the previous code and the way it worked.

The old version of commit() 'permits' occasional earlier commits of registers added by another threads as it uses the same internal variable (self.__registersList) to get the list of elements which should be send to the database. Having in mind this feature I've assumed that "an 'earlier' commit is not a problem at all" as the previous code worked this way.

Considering the order of data we also have similar behaviour in both cases. Previously, if we had 2 or more threads that were executing addRegister() simultaneously using the same client object (e.g. gDataStoreClient), we didn't have any influence on the final order of elements in the internal register (self.__registersList) and thus in database.